core/lib/zksync_core/src/api_server/web3/metrics.rs

```diff
@@ -0,0 +1,197 @@
+//! Metrics for the JSON-RPC server.
+
+use vise::{
+    Buckets, Counter, EncodeLabelSet, EncodeLabelValue, Family, Gauge, Histogram, LabeledFamily,
+    LatencyObserver, Metrics,
+};
+
+use std::{
+    fmt,
+    time::{Duration, Instant},
+};
+
+use zksync_types::api;
+
+use super::ApiTransport;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "scheme", rename_all = "UPPERCASE")]
+pub(super) enum ApiTransportLabel {
+    Http,
+    Ws,
+}
+
+impl From<&ApiTransport> for ApiTransportLabel {
+    fn from(transport: &ApiTransport) -> Self {
+        match transport {
+            ApiTransport::Http(_) => Self::Http,
+            ApiTransport::WebSocket(_) => Self::Ws,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue)]
+#[metrics(rename_all = "snake_case")] // FIXME: was PascalCase
```

Comment:
`rename_all` currently doesn't support PascalCase, and IMO it would make sense to rename these label values for consistency with other label values.